src/Haskell/Ide/Engine/Plugin/HsImport.hs

```diff
@@ -67,8 +74,34 @@ importModule uri modName =
         Nothing -> do
           newText <- liftIO $ T.readFile output
           liftIO $ removeFile output
-          workspaceEdit <- liftToGhc $ makeDiffResult input newText fileMap
-          return $ IdeResultOk workspaceEdit
+          J.WorkspaceEdit mChanges mDocChanges <- liftToGhc $ makeDiffResult input newText fileMap
+
+          shouldFormat <- hasFormattedImports input
+          if shouldFormat
+            then do
+              confFile <- liftIO $ Brittany.getConfFile origInput
+              -- Format the import with Brittany
+              newChanges <- forM mChanges $ mapM $ mapM (formatTextEdit confFile)
+              newDocChanges <- forM mDocChanges $ mapM $ \(J.TextDocumentEdit vDocId tes) -> do
+                ftes <- forM tes (formatTextEdit confFile)
+                return (J.TextDocumentEdit vDocId ftes)
+    
+              return $ IdeResultOk (J.WorkspaceEdit newChanges newDocChanges)
+            else
+              return $ IdeResultOk (J.WorkspaceEdit mChanges mDocChanges)
+
+  where hasFormattedImports fp = do
+          ls <- T.lines <$> liftIO (T.readFile fp)
```

Comment:
Unrelated to this PR, can we consider not using the read function from the Text package? I’ve had problem in the past with locale settings messing up the results. 

An alternative is to used ByteString and do a lenient decode to utf8

Reply:
Could this same issue be causing #667?

Reply:
Yes, that's exactly how the problem manifests. Should we create a ticket to get this addressed?

Reply:
@Lorenzo We can probably just reuse that ticket, and add some comments explaining that it could be related to `Data.Text.IO.readFile`. I just did a quick search and hie does use it in quite a few locations

Reply:
I ran into similar issues when working on the PureScript IDE integration, if it's any help we're using these functions to read source files currently: https://github.com/purescript/purescript/blob/b07042fb7b4f6a68c0fdc68b7e51076b7106cfba/src/System/IO/UTF8.hs